File: go/arraylist/reverse_string.go
package arraylist

func ReverseString(s []byte) {
	p1, p2 := 0, len(s)-1
	for p1 < p2 {
		s[p1], s[p2] = s[p2], s[p1]
		p1++
		p2--
	}
}

File: rust/src/arraylist/reverse_string.rs
pub fn reverse_string(s: &mut Vec<char>) {
    let mut p1 = 0_usize;
    let mut p2 = s.len()-1;
    while p1 < p2 {
        let tmp = s[p1];
        s[p1] = s[p2];
        s[p2] = tmp;
        p1 += 1;
        p2 -= 1
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_reverse_string() {
        let mut s = vec!['h', 'e', 'l', 'l', 'o'];
        reverse_string(&mut s);
        assert_eq!(vec!['o', 'l', 'l', 'e', 'h'], s)
    }
}